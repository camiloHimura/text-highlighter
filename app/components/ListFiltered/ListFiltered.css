.listFiltered{
    height: 20vh;
    padding: 6px;
    overflow: auto;
    border: 1px solid black;
}

.listFiltered > div{
    width: fit-content;
    padding: 2px 10px;
    margin-bottom: 6px;
    position: relative;
    transition: all .2s;
    overflow: hidden;
}

.listFiltered > div:hover{
    padding-left: 34px;
}

.listFiltered .close{
    top: 50%;
    left: 0px;
    width: 20px;
    height: 20px;
    border: none;
    outline: none;
    color: white;
    margin-top: -10px;
    position: absolute;
    background: black;
    transition: all .2s;
    transform: translateX(-20px)
}

.listFiltered > div:hover .close{
    transform: translateX(6px)
}