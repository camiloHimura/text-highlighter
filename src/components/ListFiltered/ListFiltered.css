.listFiltered{
    height: 20vh;
    overflow: auto;
    border: 1px solid black;
}

.listFiltered > div{
    margin-bottom: 6px;
}
