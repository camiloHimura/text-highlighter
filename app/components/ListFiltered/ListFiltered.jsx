import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import PropTypes from "prop-types";
import "./ListFiltered.css"

import {removeTextAction} from "../../state/actions";

const mapStateToProps = state => {
    return {
            textList: state.textList,
            sortAscending: state.textList.sortAscending,
        }            
}

const mapDispachToProps = dispatch => {
    return {
        removeTextAction: (text, color) => dispatch(removeTextAction({text, color})),
    }
}

function ListFiltered(props) {
    const {textList, selectFilter, sortAscending, removeTextAction} = props;
    let finalList = textList[selectFilter] || [];

    useEffect(() => {
        finalList = finalList.sort(sortBy(sortAscending))
        
    }, [finalList.length, sortAscending]);

    function sortBy(isAsending){
        return (a, b) => {
            a = a.toLocaleLowerCase();
            b = b.toLocaleLowerCase();

            if (a > b) { return isAsending ? -1: 1 }
            if (b > a) { return isAsending ? 1 : -1 }
            return 0;
        }
    }

    function removeText(text, color){
        removeTextAction(text, color)
    }

    return  <div className="listFiltered">
                {finalList.map((info, key) => {
                            return <div key={`${key}-${info}${selectFilter}`}style={{"background": selectFilter}}>
                                        <span>{info}</span>
                                        <button className="close" onClick={() => removeText(info, selectFilter)}>x</button>
                                    </div>
                        })
                }
            </div>

}

ListFiltered.propTypes = {
    textList: PropTypes.shape({
        sortAscending: PropTypes.bool.isRequired
    }),
    selectFilter: PropTypes.string.isRequired, 
    sortAscending: PropTypes.bool.isRequired, 
    removeTextAction: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispachToProps)(ListFiltered);
