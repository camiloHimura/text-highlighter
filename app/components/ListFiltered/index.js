import ListFiltered from "./ListFiltered.jsx";
export default ListFiltered;