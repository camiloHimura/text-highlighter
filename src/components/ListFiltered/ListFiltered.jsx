import React from 'react';
import {connect} from 'react-redux';
import "./ListFiltered.css"


const mapStateToProps = state => {
    return {
            textList: state.textList
        }            
}

function ListFiltered(props){
    console.log(props)
    const {textList, selectFilter} = props;
    console.log(textList[selectFilter])
    const finalList = textList[selectFilter] || [];

    return  <div className="listFiltered">
                {finalList.map((info, key) => {
                            return <div key={`${key}-${info}${selectFilter}`}style={{"background": selectFilter}}>
                                        {info}
                                    </div>
                        })
                }
            </div>

}

export default connect(mapStateToProps)(ListFiltered);
