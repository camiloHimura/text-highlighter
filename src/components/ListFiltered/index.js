import ListFiltered from "./ListFiltered";
export default ListFiltered;