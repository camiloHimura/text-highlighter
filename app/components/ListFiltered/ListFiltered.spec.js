import React from 'react';
import { ListFiltered } from './ListFiltered';
import { shallow, mount } from 'enzyme'


describe('ListFiltered component', () => {
  let Component;
  let props;

  beforeEach(() => {
    props = {
      textList: {
        red: ['abc', 'bcd', 'cda']
      },
      selectFilter: 'red', 
      sortAscending: false, 
      removeTextAction: jest.fn()
    }
    Component = mount(<ListFiltered {...props}/>);
  })
  it('take snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
  
  it('list rendered', () => {
    const containers = Component.find('[data-test="contText"]');
    expect(containers).toHaveLength(props.textList[props.selectFilter].length);
    containers.forEach((node, index) => {
      expect(node.find('span').text()).toEqual(props.textList[props.selectFilter][index])
      expect(node.prop('style')).toEqual({"background": props.selectFilter})
    })
  });
  
  it('sorting list Ascending', () => {
    Component.setProps({ sortAscending: true });
    const containers = Component.find('[data-test="contText"]');
    // invertes test's list is 'assending' order
    const ascendingValues = [...props.textList[props.selectFilter]].reverse();
    containers.forEach((node, index) => {
      expect(node.find('span').text()).toEqual(ascendingValues[index])
    })
  });

})