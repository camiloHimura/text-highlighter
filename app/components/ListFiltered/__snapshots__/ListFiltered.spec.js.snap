// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFiltered component take snapshot 1`] = `
<ListFiltered
  removeTextAction={[MockFunction]}
  selectFilter="red"
  sortAscending={false}
  textList={
    Object {
      "red": Array [
        "abc",
        "bcd",
        "cda",
      ],
    }
  }
>
  <styled.div
    className="listFiltered"
  >
    <StyledComponent
      className="listFiltered"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "iwKOFX",
            "rules": Array [
              "
  height: 20vh;
  padding: 6px;
  overflow: auto;
  border: 1px solid black;
  
  & > div{
    width: fit-content;
    padding: 2px 10px;
    margin-bottom: 6px;
    position: relative;
    transition: all .2s;
    overflow: hidden;
    
    &:hover{
      padding-left: 34px;
    
      & .close{
        transform: translateX(6px)
      }
    }
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa iwKOFX listFiltered"
      >
        <div
          data-test="contText"
          key="0-abcred"
          style={
            Object {
              "background": "red",
            }
          }
        >
          <span>
            abc
          </span>
          <styled.button
            className="close"
            onClick={[Function]}
          >
            <StyledComponent
              className="close"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "cEdXND",
                    "rules": Array [
                      "
  top: 50%;
  left: 0px;
  width: 20px;
  height: 20px;
  border: none;
  outline: none;
  color: white;
  margin-top: -10px;
  position: absolute;
  background: black;
  transition: all .2s;
  transform: translateX(-20px)
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <button
                className="sc-bwzfXH cEdXND close"
                onClick={[Function]}
              >
                x
              </button>
            </StyledComponent>
          </styled.button>
        </div>
        <div
          data-test="contText"
          key="1-bcdred"
          style={
            Object {
              "background": "red",
            }
          }
        >
          <span>
            bcd
          </span>
          <styled.button
            className="close"
            onClick={[Function]}
          >
            <StyledComponent
              className="close"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "cEdXND",
                    "rules": Array [
                      "
  top: 50%;
  left: 0px;
  width: 20px;
  height: 20px;
  border: none;
  outline: none;
  color: white;
  margin-top: -10px;
  position: absolute;
  background: black;
  transition: all .2s;
  transform: translateX(-20px)
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <button
                className="sc-bwzfXH cEdXND close"
                onClick={[Function]}
              >
                x
              </button>
            </StyledComponent>
          </styled.button>
        </div>
        <div
          data-test="contText"
          key="2-cdared"
          style={
            Object {
              "background": "red",
            }
          }
        >
          <span>
            cda
          </span>
          <styled.button
            className="close"
            onClick={[Function]}
          >
            <StyledComponent
              className="close"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "cEdXND",
                    "rules": Array [
                      "
  top: 50%;
  left: 0px;
  width: 20px;
  height: 20px;
  border: none;
  outline: none;
  color: white;
  margin-top: -10px;
  position: absolute;
  background: black;
  transition: all .2s;
  transform: translateX(-20px)
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <button
                className="sc-bwzfXH cEdXND close"
                onClick={[Function]}
              >
                x
              </button>
            </StyledComponent>
          </styled.button>
        </div>
      </div>
    </StyledComponent>
  </styled.div>
</ListFiltered>
`;
